raiden/transfer/state.py

```diff
@@ -94,6 +93,10 @@ class RoutesState(State):
     )
 
     def __init__(self, available_routes):
+        # consume possible generators and make a copy of the routes since the
+        # tasks will modify this list in-place
```

Comment:
Since we don't have immutable data types in python, at least not in the standard library and I didn't want to add a new dependency that did a bunch o magic, at some point in time we need to modify the state objects in place, that's why we make a [deepcopy on the dispatch](https://github.com/raiden-network/raiden/blob/master/raiden/transfer/architecture.py#L125)